packages/sdk/src/types/runtime/actor.ts

```diff
@@ -437,6 +439,59 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
         return null;
     }
 
+    /**
+     * Starts playing a preloaded sound.
+     * @param soundAssetId Name of sound asset preloaded using AssetManager.
+     * @param options Adjustments to pitch and volume, and other characteristics.
+     * @param startTimeOffset How many seconds to offset into the sound
+     */
+    public playSound(
+        soundAssetId: string, options: SetSoundStateOptions,
+        startTimeOffset?: number): ForwardPromise<SoundInstance> {
+
+        return createForwardPromise(new SoundInstance("SoundAssetHandle"),
+            new Promise<SoundInstance>((resolve, reject) => {
+                this.created().then(() => {
+                    this._context.assetManager.SoundLoaded(soundAssetId).then(() => {
+
+                        this.context.internal.protocol.sendPayload(
+                            {
+                                type: 'play-sound',
+                                actorId: this.id,
+                                soundAssetId,
+                                startTimeOffset
+                            } as PlaySound);
+                        resolve();
+                    }).catch((reason: any) => {
+                        log.error('app', `Failed PlaySound on actor ${this.id}. ${(reason || '').toString()}`.trim());
+                        reject();
+                    });
+                }).catch((reason: any) => {
+                    log.error('app', `Failed PlaySound on actor ${this.id}. ${(reason || '').toString()}`.trim());
+                    reject();
+                });
+            })
+        );
+        return;
+    }
+
+    /**
+     * Loads a sound file and starts playing.
+     * @param uri Path to a sound asset that needs to be loaded and played.
+     * @param options Adjustments to pitch and volume, and other characteristics.
+     * @param startTimeOffset How many seconds to offset into the sound
+     */
+    public loadAndPlaySound(
+        uri: string, loadOptions: LoadSoundOptions,
+        playOptions: SetSoundStateOptions,
+        startTimeOffset?: number): ForwardPromise<SoundInstance> {
+        return createForwardPromise(new SoundInstance("SoundAssetHandle"),
+            new Promise<SoundInstance>((resolve, reject) => {
+                resolve();
+            })

```

Comment:
Could be replaced with `Promise.resolve(soundInstance)`

Reply:
this will be completely rewritten, this does nothing

---
In reply to: [258164784](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/204#discussion_r258164784) [](ancestors = 258164784)